Source/Core/Core/ConfigLoaders/BaseConfigLoader.cpp

```diff
@@ -53,7 +54,19 @@ class INIBaseConfigLayerLoader final : public Config::ConfigLayerLoader
         const IniFile::Section::SectionMap& section_map = section.GetValues();
 
         for (const auto& value : section_map)
-          config_section->Set(value.first, value.second);
+        {
+          const Config::ConfigLocation location{system.first, section_name, value.first};
```

Comment:
This logic is repeated multiple times (in two files), would it make sense to move it to a shared common function?